src/core/fetchers/segment/segment_fetcher.ts

```diff
@@ -31,7 +31,8 @@ import {
 import { formatError } from "../../../errors";
 import { ISegment } from "../../../manifest";
 import {
-  ISegmentParserResponse,
+  ISegmentParserInitSegment,
+  ISegmentParserSegment,
```

Comment:
I noticed that the `ISegmentParserResponse` type was only used here, and felt it was complexifying types by adding one level of indirection, so I just removed it.